### PR TITLE
Add mongodb support for kofu

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

File: kofu/build.gradle.kts [hunk 1]
```diff
@@ -50,6 +50,7 @@ dependencies {
 	testCompile("org.testcontainers:testcontainers:1.11.3")
 	testImplementation("redis.clients:jedis")
 	testImplementation("io.lettuce:lettuce-core")
+	testImplementation("org.mongodb:mongodb-driver-legacy")
```
Comment (sdeleuze) [inline kofu/build.gradle.kts:53]: Could you please elaborate on this `mongodb-driver-legacy` dependency?

File: kofu/src/main/kotlin/org/springframework/fu/kofu/mongo/AbstractMongoDsl.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoInitializer
+import org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoProperties
+import org.springframework.context.support.GenericApplicationContext
+import org.springframework.fu.kofu.AbstractDsl
+import org.springframework.fu.kofu.ConfigurationDsl
+
+/**
+ * Kofu DSL for MongoDB configuration.
+ *
+ * Enable and configure Reactive MongoDB support by registering a [org.springframework.data.mongodb.core.MongoTemplate].
```
Comment (sdeleuze) [inline kofu/src/main/kotlin/org/springframework/fu/kofu/mongo/AbstractMongoDsl.kt:33]: Likely not a Reactive variant, right?

File: kofu/build.gradle.kts (hunk 1 above)
Comment (eddumelendez) [reply kofu/build.gradle.kts:53]: in order to run the test, `MongoInitializer` needs `com.mongodb.client.MongoClient` from that dependency.

File: kofu/src/main/kotlin/org/springframework/fu/kofu/mongo/AbstractMongoDsl.kt (hunk 2 above)
Comment (eddumelendez) [reply kofu/src/main/kotlin/org/springframework/fu/kofu/mongo/AbstractMongoDsl.kt]: oops. fixed!

**Comment by fteychene** [comment]
I adopted the PR to fix merge conflict and change the impacts from the migration to boot 2.4.0

**Comment by fteychene** [comment]
LGTM :rocket:

**Pull request merged**